docs/reference/ingest/processors/inference.asciidoc

```diff
@@ -0,0 +1,77 @@
+[role="xpack"]
+[testenv="basic"]
+[[inference-processor]]
+=== {infer-cap} Processor
+
+Uses a pre-trained {dfanalytics} model to infer against the data that is being 
+ingested in the pipeline.
+
+
+[[inference-options]]
+.{infer-cap} Options
+[options="header"]
+|======
+| Name               | Required  | Default                        | Description
+| `model_id   `      | yes       | -                              | (String) The ID of the model to load and infer against.
+| `target_field`     | yes       | `ml.inference.<processor_tag>` | (String) The field name to which to write the results objects.
+| `field_mappings`   | no        | -                              | (Object) Maps the document field names to the known field names of the model.
+| `inference_config` | yes       | -                              | (Object) Contains the inference type and its options. There are two types: `regression` and `classification`.
```

Comment:
It would probably be helpful to have full docs referencing all the options for `regression` and `classification` configs. I am not sure where they would fit in this doc format (probably outside the processor field table though).

```
regression:
- results_field: the field to which the inference prediction is written. Defaults to `predicted_value`
```
```
classification:
- results_field: the field to which the inference prediction is written. Defaults to predicted_value
- top_num_classes: The number of top class predictions to return. Defaults to 0.
- top_classes_results_field: The field to which the top classes are written. Defaults to top_classes
```